src/ucp/rndv/proto_rndv.c

```diff
@@ -761,3 +765,72 @@ void ucp_proto_rndv_bulk_request_init_lane_idx(
 
     req->send.multi_lane_idx = lane_idx - 1;
 }
+
+static ucs_status_t
+ucp_proto_rndv_ats_init(const ucp_proto_init_params_t *params)
+{
+    ucp_proto_perf_type_t perf_type;
+    size_t max_length;
+
+    if (ucp_proto_rndv_init_params_is_ppln_frag(params)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    /* This protocols supports either a regular rendezvous receive but without
+     * data, or a rendezvous receive which should ignore the data.
+     * In either case, we just need to send an ATS.
+     */
+    if (params->select_param->op_id == UCP_OP_ID_RNDV_RECV) {
+        max_length = 0;
+    } else if (params->select_param->op_id == UCP_OP_ID_RNDV_RECV_DROP) {
+        max_length = SIZE_MAX;
+    } else {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    /* Support only 0-length messages */
+    *params->priv_size                 = sizeof(ucp_proto_rndv_ack_priv_t);
+    params->caps->cfg_thresh           = 0;
+    params->caps->cfg_priority         = 1;
+    params->caps->min_length           = 0;
+    params->caps->num_ranges           = 1;
+    params->caps->ranges[0].max_length = max_length;
+    params->caps->ranges[0].node       = NULL;
+    for (perf_type = 0; perf_type < UCP_PROTO_PERF_TYPE_LAST; ++perf_type) {
+        params->caps->ranges[0].perf[perf_type] = UCS_LINEAR_FUNC_ZERO;
+    }
+
+    return ucp_proto_rndv_ack_init(params, params->priv);
+}
+
+static ucs_status_t ucp_proto_rndv_ats_complete(ucp_request_t *req)
+{
+    if (req->send.rndv.rkey != NULL) {
+        ucp_proto_rndv_rkey_destroy(req);
+    }
+
+    ucp_datatype_iter_cleanup(&req->send.state.dt_iter, UCP_DT_MASK_ALL);
+
+    return ucp_proto_rndv_recv_complete(req);
+}
+
+static ucs_status_t
+ucp_proto_rndv_ats_proto_progress(uct_pending_req_t *uct_req)
+{
+    ucp_request_t *req = ucs_container_of(uct_req, ucp_request_t, send.uct);
+
+    return ucp_proto_rndv_ack_progress(req, req->send.proto_config->priv,
```

Comment:
if we want to make it more similar to other rndv protocol such as rndv_get, we could do something like
so ucp_proto_rndv_ats_complete will destroy rkey and dt->finish, and move to STAGE_ATS
```
.progress = {
    [STAGE_INIT] = ucp_proto_rndv_ats_complete,
    [STAGE_ATS] =  ucp_proto_rndv_ats_proto_progress,
}
```

Reply:
i made it as a part of request initialization. Imo no need for a separate stage which does not send (or wait) anything